controlplane/pkg/nsm/nsm.go

```diff
@@ -891,7 +932,10 @@ func (srv *networkServiceManager) RemoteConnectionLost(clientConnection nsm.NSMC
 }
 
 func (srv *networkServiceManager) closeEndpoint(ctx context.Context, clientConnection *model.ClientConnection) error {
-
+	if clientConnection.Endpoint == nil {
+		logrus.Infof("No need to close, since NSE is we know is dead at this point.")
+		return nil
+	}
```

Comment:
If `clientConnection` will be `nil` it will panic with `nil pointer dereference`. We should check that before refering to `clientConnection.Endpoint` I think.

Reply:
It is used only from places there it is not nil in any case.